app/services/credit_notes/create_from_progressive_billing_invoice.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module CreditNotes
+  class CreateFromProgressiveBillingInvoice < BaseService
+    def initialize(progressive_billing_invoice:, amount:, reason: :other)
+      @progressive_billing_invoice = progressive_billing_invoice
+      @amount = amount
+      @reason = reason
+
+      super
+    end
+
+    def call
+      return result unless amount.positive?
+      return result.forbidden_failure! unless progressive_billing_invoice.progressive_billing?
+
+      # Important to call this method as it modifies @amount if needed
+      items = calculate_items!
+
+      CreditNotes::CreateService.new(
+        invoice: progressive_billing_invoice,
+        credit_amount_cents: amount,
```

Comment:
In this case, the amount must take taxes into account.
You can check for `CreditNotes::CreateFromTermination#creditable_amount_cents` that applies a similar logic